DESIGN.md

```diff
@@ -54,39 +56,113 @@ Usage:
+These commands are specifically limited to those specified here and relate to the principle of least astonishment.
+These are the types of commands users will be used to seeing in other database shells. These aren't queries and cannot
+be executed by the database query engine, rather they are commands at the level of the database itself, rather than
+the data in the database.
+
+* `start transaction` or `begin`
+  * This starts a transaction. If we are in auto-commit mode, the shell reports a warning saying that this has no 
+    effect. 
+* `commit`
+  * This commits a transaction. If there is no transaction in progress, the shell reports an error saying that there is 
+    no active transaction. If we are in auto-commit mode, the command is accepted and no error or warning or information
+    is reported.
+* `use [ledger|endpoint] ledger-or-endpoint`
+  * Connect to a different endpoint or ledger.
+* `show [ledgers|tables]`
+  * Display a list of tables in the current ledger, or ledgers at the current endpoint, depending on the command 
+    arguments.
+* `help`
+  * Prints the lists of database and meta commands.
+* `quit`
+  * Quits the shell.
+
+### Shell Meta Commands
 
 All commands to the shell itself will be prefixed with a backslash (\), e.g:
 
 * `\quit`
-    * Quits the shell
+  * Quits the shell. This can be preceded by the backslash or not.
 * `\help`
-    * Prints the lists of meta commands
+  * Prints the lists of meta commands. This can be preceded by the backslash or not.
 * `\status`
-    * Prints out things like connection status, server ping latency, etc.
+  * Prints out things like connection status, server ping latency, etc.
 * `\metrics [on|off|last]`
-    * Determines whether or not metrics will be emitted after the results of a query are shown.
-    * last prints out the last known metrics from your previous command.
-* `\query-results-format [ion|json|table]`
-    * ion Prints the objects from the database as ION documents in text.
-    * json Prints the objects from the database as JSON documents in text.
-    * table Tabulates the data and prints out the data as rows.
+  * Determines whether or not metrics will be emitted after the results of a query are shown.
+  * last prints out the last known metrics from your previous command.
+* `\query-results-format [ion|json|table|csv]`
+  * ion Prints the objects from the database as ION documents in text.
+  * json Prints the objects from the database as JSON documents in text.
+  * table Tabulates the data and prints out the data as rows.
+  * table Prints out the data in CSV format.
 * `\auto-commit`
-    * Determines whether each statement will be executed as a transaction or
-      not. By default this is off, meaning that statements are all executed as
-      individual transactions.
+  * Determines whether each statement will be executed as a transaction or
+    not. By default this is off, meaning that statements are all executed as
+    individual transactions.
```

Comment:
I've been thinking about this. Do we want to have first-class support for every setting or do we want a generic `set opt=val` where opt and val could also be provided in the config or CLI (via `--opt opt=val`)?

Reply:
Agree generally. I think auto-commit is important enough to be top-level though... again, no strong opinion here.

Reply:
The question to me is whether it's even worth thinking about it. For example, with openssh it says

```
     -o option
             Can be used to give options in the format used in the configuration file.  This is useful for specifying options for which there is no separate command-line flag.  For full details of the options
             listed below, and their possible values, see ssh_config(5).
```

I think that's a nice model, and simple. If it's a setting in the configuration file, then you can override it with `-o opt=val`.

Reply:
Fixed